gnovm/pkg/gnolang/op_binary.go

```diff
@@ -344,6 +345,9 @@ func isEql(store Store, lv, rv *TypedValue) bool {
 	} else if rvu {
 		return false
 	}
+	if !isSameType(lv.T, rv.T) {
+		return false
```

Comment:
What cases does it handle? Does it make more sense to panic saying the types are incompatible or is this done elsewhere?

Reply:
It's for this case, where `Error(0)` and `errCmp` and both comparable and `Error(0)` is assignable to `errCmp`, that makes the comparison works and get a false result since the actually have different concrete types.

```go
package main

import (
    "errors"
    "strconv"
)

type Error int64

func (e Error) Error() string {
    return "error: " + strconv.Itoa(int(e))
}

var errCmp = errors.New("XXXX")

func main() {
    if Error(0) == errCmp {
        println("what the firetruck?")
    } else {
        println("something else")
    }
}
```

Reply:
this logic has been moved to #1890.